src/contracts/Staking.sol

```diff
@@ -419,20 +423,45 @@ contract Staking is Ownable {
     }
 
     /**
-        @notice redeem REWARD_TOKEN for STAKING_TOKEN instantly with fee
-        @param _amount uint
+        @notice redeem REWARD_TOKEN for STAKING_TOKEN instantly with fee.  Must use entire amount
+        @notice this is in the staking contract due to users having reward tokens (potentially) in the warmup contract
         @param _trigger bool
      */
 
-    function instantUnstake(uint256 _amount, bool _trigger) external {
+    function instantUnstake(bool _trigger) external {
+        // prevent unstaking if override due to vulnerabilities
         require(!pauseUnstaking, "Unstaking is paused");
         if (_trigger) {
             rebase();
         }
-        _retrieveBalanceFromUser(_amount, msg.sender);
+
+        Claim memory userWarmInfo = warmUpInfo[msg.sender];
+
+        uint256 walletBalance = IERC20(REWARD_TOKEN).balanceOf(msg.sender);
+        uint256 warmUpBalance = IRewardToken(REWARD_TOKEN).balanceForGons(
```

Comment:
I am still unsure as to why the locks exist on the claims, but doesn't this break that functionality since we don't check it here?